src/main/java/org/jenkinsci/test/acceptance/po/Jenkins.java

```diff
@@ -112,7 +116,15 @@ public void restart() {
         visit("restart");
         clickButton("Yes");
 
-        waitFor(driver, not(hasContent("Please wait")), JenkinsController.STARTUP_TIMEOUT);
+        try {
+            waitFor(driver, not(hasContent("Please wait")), JenkinsController.STARTUP_TIMEOUT);
+        }catch(TimeoutException e) {
+            //Let's try to avoid false negatives or not auto refresh
+            visit(driver.getCurrentUrl());
+            //we wait 10 seconds for refresh things.
+            //breadcrumbBarAnchor is only present when the Jenkins is ready
```

Comment:
And can I assume that is present even when there is no anonymous read access and you have been redirected to the login screen?